src/playdate/util/sparsemap.nim

```diff
@@ -0,0 +1,98 @@
+type
+    Pair*[K, V] = tuple[key: K, value: V]
+
+    Entry[K, V] = object
+        pair: Pair[K, V]
+        sparseIdx: uint32
+
+    DenseIdx = distinct uint32
+
+    StaticSparseMap*[N : static int, K, V] {.byref.} = object
```

Comment:
Im probably not familiar enough with sparse maps, but I don't know what N is. Capacity?

Could do with a comment

Reply:
Good call. I updated the documentation throughout